pom.xml

```diff
@@ -354,14 +354,15 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-shade-plugin</artifactId>
-                    <version>3.2.1</version>
+                    <version>3.2.4</version>
                     <executions>
                         <execution>
                             <phase>package</phase>
                             <goals>
                                 <goal>shade</goal>
                             </goals>
                             <configuration>
+                                <minimizeJar>true</minimizeJar>
```

Comment:
Preparing to be pleasantly surprised, but as a warning, I think I tried this once and it died because it couldn't figure out a bunch of stuff that was loaded via reflection and the like.

Reply:
i posted the results in the PR original comment

Reply:
Yeah, to elaborate. I think it removes too much because it can't figure out everything that is being loaded in a static way in only java code. For example, this https://stackoverflow.com/questions/8817257/minimize-an-uber-jar-correctly-using-shade-plugin notes that some classes are specified in properties files but also notes you can whitelist certain dependencies.